compiler-core/src/language_server/engine.rs

```diff
@@ -205,6 +205,17 @@ where
             let byte_index =
                 line_numbers.byte_index(params.position.line, params.position.character);
 
+            // Do not suggest in comments
+            if module.extra.is_within_comment(byte_index)
+                || (byte_index > 0
+                    && module
+                        .extra
+                        // Cursor is at the end of the comment
+                        .is_within_comment(byte_index - 1))
```

Comment:
How come this is done here and not within `is_within_comment`?

Reply:
`is_within_comment` simply checks the byte index is in the range `[span.start, span.end)` in any comment.

For example:

```gleam
// comment
          ^cursor is at byte 10 in insert mode (after `t`)
```

The comment above has the span of `[2, 10)`, then `is_within_comment(10)` here returns false. So to ensure it knows it is in the comment context, we `-1` it.

For completeness, only using `byte_index - 1` doesn't cover the case when the following happens:

```gleam
//
  ^cursor is at byte 2 (after the last `/`)
```

The above has the span of `[2, 3)`, then `is_within_comment(2 - 1)` here returns false. So we don't `-1` in this case.

Reply:
Right, so why isn't this done inside the method? Seems like without that it doesn't do what it claims to do. Is that right?

Reply:
> Right, so why isn't this done inside the method? Seems like without that it doesn't do what it claims to do. Is that right?

`is_within_comment` is working as intended, which is to check if the byte index is in a comment.

If we try to incorporate this `-1` adjustment into the method, we will break things. 

Suppose we have an empty comment, then followed by a newline character (`//\n`). If we have a byte index that points to the newline (2 in this case, in vim's normal mode), we shouldn't include the newline in the comment. This means that we shouldn't `-1` the byte index in the method.


For reference, changing from normal to insert mode in the last character of the line bumps the offset number as shown below,

<img width="431" alt="image" src="https://github.com/gleam-lang/gleam/assets/65026286/31da0a46-6b44-47ce-b689-f798bf15b18d">

<img width="420" alt="image" src="https://github.com/gleam-lang/gleam/assets/65026286/1a044768-6da9-43cd-b058-140d25f7aea3">

Reply:
Ignore this section for now, I've found another uncovered edge case on empty comments.

Reply:
I don't understand why the logic for whether a byte index is inside a comment is different here than in other situations. Could you please explain with examples for both situations. Thank you